src/Pair.sol

```diff
@@ -73,7 +71,7 @@ abstract contract Pair is IPair, UniswapV2ERC20 {
         emit CustomSwapFeeChanged(customSwapFee, _customSwapFee);
         customSwapFee = _customSwapFee;
 
-        updateSwapFee();
+        this.updateSwapFee();
```

Comment:
Why use `CALL` instead of just an abstract function?

Reply:
yeah so it's either this way or we have to define the abstract function in `Pair.sol` and then we have to do the `override` in `ConstantProductPair / StablePair`

```solidity 
    function updateSwapFee() public override(IPair, Pair) 
```

using `this` doesn't make it an external call does it?

Reply:
It does make it an external call, debug and take a look if you want (or easier, run with `-vvvv`)

Reply:
okay indeed it was an external call. With a gas cost of `3880`. So we won't do that.